Automattic/woocommerce-payments
# Tax is calculated twice for virtual products if Apple/Google Pay is used on product pages

Fixes #7577

#### Changes proposed in this Pull Request

- This PR renames the method `prices_exclude_tax` to `cart_prices_include_tax` and invert its logic to make the code easier to read.
- It also adds two tests to ensure tax calculation is correct and consistent for product pages and cart details
- The key fix was changing `$product->get_price()` to `wc_get_price_excluding_tax()`
- WooPay class was also updated for consistency





#### Testing instructions

1. Setup Apple Pay in your environment. pdjVC1-11t-p2
1. Go to WooCommerce >> Settings >> Tax and set the settings from the original issue:
	1. Yes, I will enter prices inclusive of tax
	2. Shop base address (for consistency, https://github.com/Automattic/woocommerce-payments/issues/7577#issuecomment-1847967971)
	3. Display prices in shop **Including taxes**
	4. Display prices during cart and checkout **Excluding taxes**
1. Add a Standard tax rate
2. Try to use Apple Pay and ensure the price is correct

- Additionally, run other scenarios from the spreadsheet from https://github.com/Automattic/woocommerce-payments/issues/7577#issuecomment-1847967971 and ensure everything works as expected.

-------------------


**Post merge**
